services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ReceiveMessageBatchManager.java

```diff
@@ -91,9 +91,6 @@ private String checkBatchingEligibility(ReceiveMessageRequest rq) {
         if (rq.overrideConfiguration().isPresent()) {
             return "Request has override configurations.";
         }
-        if (rq.waitTimeSeconds() != null && rq.waitTimeSeconds() != 0) {
```

Comment:
Why are we removing this?

Reply:
If the user passes this value then user can control the Polling time out , this is behaviour same as https://github.com/aws/aws-sdk-java/blob/61d73631fac8535ad70666bbce9e70a1d2cea2ca/aws-java-sdk-sqs/src/main/java/com/amazonaws/services/sqs/buffered/ReceiveQueueBuffer.java#L192-L197

I removed this in earlier PR but when I was doing V1 comparison I realized that V1 uses this value to determine the Client side Wait time for that  particular request.

Reply:
We already supported in V2 https://github.com/aws/aws-sdk-java-v2/blob/dd9379d67d08d932eeda472e6616e4e06b893bcd/services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/QueueAttributesManager.java#L76-L82
But after handling the Surface API review comments where I added a check to bypass, I went ahead and added check to this since it was an additional Attribute related to long polling but when used with Automatic Batch manager we use it client side polling time so thus we need this to go through the batch manager